services/user/Evaluations/query-service/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+#[psibase::service]
+#[allow(non_snake_case)]
+mod service {
+    use async_graphql::connection::Connection;
+    use async_graphql::*;
+    use evaluations::service::{
+        ConfigTable, Evaluation, EvaluationTable, Group, GroupTable, User, UserSettings,
+        UserSettingsTable, UserTable,
+    };
+    use psibase::*;
+    use serde::Deserialize;
+    use serde_aux::field_attributes::deserialize_number_from_string;
+
+    #[derive(Deserialize, SimpleObject)]
+    struct KeysSet {
+        owner: AccountNumber,
+        evaluation_id: String,
+        group_number: String,
+        keys: Vec<Vec<u8>>,
+        hash: String,
+    }
+
+    #[derive(Deserialize, SimpleObject)]
+    struct GroupFinish {
+        owner: AccountNumber,
+        #[serde(deserialize_with = "deserialize_number_from_string")]
+        evaluation_id: u32,
+        #[serde(deserialize_with = "deserialize_number_from_string")]
+        group_number: u32,
+        users: Vec<AccountNumber>,
+        result: Vec<u8>,
+    }
+
+    struct Query;
+
+    #[Object]
+    impl Query {
+        async fn get_group_key(
+            &self,
+            evaluation_owner: AccountNumber,
+            evaluation_id: u32,
+            group_number: u32,
+        ) -> async_graphql::Result<Connection<u64, KeysSet>> {
+            EventQuery::new("history.evaluations.keysset")
+                .condition(format!(
+                    "owner = '{}' AND evaluation_id = {} AND group_number = {}",
```

Comment:
Actually I think that the `evaluation_id = {} AND group_number = {}` needs to quote the numbers like `evaluation_id = '{}' AND group_number = '{}'`

Same comment for `get_group_result`. This is because of what we rediscovered yesterday about numbers being strings  in the event database (SQLite db)

Reply:
I was getting errors when I attempted this, will check it over and determine if I misconfigured something else, however, it's currently working as is with all queries working as expected.

Reply:
Integers are only turned into strings when generating the final JSON result. They're still i64 in the query.